app.js

```diff
@@ -77,25 +78,112 @@ app.get('/related-document-information', async function( req, res ) {
-    const nTriples = triples.map(t => serializeTriple(t)) || [];
-
-    res.set('Content-Type', 'text/turtle');
-    return res.send(nTriples.join('\n'));
+    return sendTurtleResponse(res, triples);
   }
   catch (error) {
     return res.status(500).json({ error: error.message });
   }
 });
+
+app.get('/search-documents', async function (req, res) {
+  try {
+    const forDecisionType = req.query.forDecisionType;
+    const forEenheid = req.query.forEenheid;
+
+    if (!forDecisionType || !forEenheid) {
+      return res.status(400).json({
+        error: "Missing required query parameters. Please provide 'forDecisionType' and 'forEenheid'."
+      });
+    }
+
+    const fromEenheid = await bestuurseenheidForSession(req.sessionUri);
+
+    if (!fromEenheid) {
+      return res.status(400).json({
+        error: "No eenheid found for mu-session-id. Aborting"
+      });
+    }
+
+    let query;
+
+    if (await isCKB(fromEenheid)) {
+      const relatedDecisionType = ckbDecisionTypeToRelatedType(forDecisionType);
+
+      if (!relatedDecisionType) {
+        return invalidDecisionTypeError(res, forDecisionType);
+      }
+
+      query = prepareCKBSearchQuery({ fromEenheid, forEenheid, decisionType: relatedDecisionType });
+    } else {
+      // Figure out whether Eenheid is related to CKB
+      let ckbUri = await getRelatedToCKB(forEenheid);
+
+      if (BYPASS_HOP_CENTRAAL_BESTUUR) {
+        console.warn(`Skipping extra hop centraal bestuur. This should only be used in development mode.`);
+        ckbUri = null;
+      }
+
+      // Get decision type to request
+      const decisionTypeData = getRelatedDecisionType(forDecisionType, ckbUri);
+
+      if (!decisionTypeData.decisionType) {
+        return invalidDecisionTypeError(res, forDecisionType);
+      }
+
+      query = prepareQuery({ fromEenheid, forEenheid, ckbUri, decisionTypeData });
+    }
+
+    // execute query
+    // TODO: Here we could add a hook to connect to vendor-API if we need to.
+    const triples = (await querySudo(query))?.results?.bindings || [];
+    return sendTurtleResponse(res, triples);
+  } catch (error) {
+    return res.status(500).json({ error: error.message });
+  }
+});
+
+app.get('/document-information', async function (req, res) {
+  try {
+    const forDecisionType = req.query.forDecisionType;
+    const forDecision = req.query.forRelatedDecision;
+
+    if (forDecision && !forDecisionType) {
+      return res.status(400).json({
+        error: `Missing required query parameters. Both "forDecision" and "forDecisionType" are required.`
+      });
+    }
+
+    const eenheid = await getEenheidForDecision(forDecision);
+    let ckbUri = await getRelatedToCKB(eenheid);
+
+    if (BYPASS_HOP_CENTRAAL_BESTUUR) {
+      console.warn(`Skipping extra hop centraal bestuur. This should only be used in development mode.`);
+      ckbUri = null;
+    }
+
+    const decisionTypeData = getRelatedDecisionType(forDecisionType, ckbUri);
```

Comment:
This won't return a result for the CKB types yet since we're not using the new mapping, but it still works since it's not really needed for the query.

If we want to lookup data for a specific decision / document, don't we only need the uri? I don't think we need the decision type and ckbUri?

I think we can simplify this and only require the `forDecision` query param?